src/dodal/devices/fast_grid_scan.py

```diff
@@ -163,53 +178,47 @@ class PandAGridScanParams(GridScanParamsCommon):
 
 
 class MotionProgram(Device):
-    def __init__(self, prefix: str, name: str = "") -> None:
+    def __init__(self, prefix: str, name: str = "", has_prog_num=True) -> None:
         super().__init__(name)
         self.running = epics_signal_r(int, prefix + "PROGBITS")
-        self.program_number = epics_signal_r(float, prefix + "CS1:PROG_NUM")
+        if has_prog_num:
+            self.program_number = epics_signal_r(float, prefix + "CS1:PROG_NUM")
+        else:
+            # Prog number PV doesn't currently exist for i02-1
+            self.program_number = soft_signal_r_and_setter(float, -1)[0]
 
 
 class FastGridScanCommon(StandardReadable, Flyable, ABC, Generic[ParamType]):
-    """Device for a general fast grid scan
+    """Device containing the minimal signals for a general fast grid scan.
 
     When the motion program is started, the goniometer will move in a snake-like grid trajectory,
-    with X as the fast axis and Y as the slow axis. If Z steps isn't 0, the goniometer will
-    then rotate in the omega direction such that it moves from the X-Y, to the X-Z plane then
-    do a second grid scan. The detector is triggered after every x step.
-    See https://github.com/DiamondLightSource/hyperion/wiki/Coordinate-Systems for more
+    with X as the fast axis and Y as the slow axis.
+
+    See ZebraFastGridScanThreeD as an example of how to implement.
     """
 
     def __init__(self, prefix: str, smargon_prefix: str, name: str = "") -> None:
```

Comment:
Could: Instead of this calling `_create_motion_program` and `_create_position_counter` on the children you could get the children to pass in the correct `MotionProgram` and `scan_invalid` signal as part of the constructor? I think it makes things a bit more simple